provisioning/device/test/_symmetric_key_registration_test.js

```diff
@@ -0,0 +1,147 @@
+
+var fakeProvisioningHost = 'fakeHost';
+var fakeIdScope = 'fakeIdScope';
+var fakeRegistrationId = 'fakeRegistrationId';
+var fakeSasToken = 'a fake token';
+
+var fakeResponse = {
+  assignedHub: 'fakeHub',
+  deviceId: 'fakeDeviceId'
+};
+
+describe('Symmetric Key Registration', function () {
+  describe('#register', function () {
+
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_006: [ `register` shall call the `getRegistrationId` method on the security object to acquire the registration id. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_008: [ `register` shall invoke `createAuthenticationToken` method on the security object to acquire a sas token object. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_004: [ `register` shall pass the SAS token into the `setSasToken` method on the transport. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_005: [ `register` shall call `register` on the polling state machine object. ] */
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_011: [ Otherwise `register` shall invoke the `_callback` with the resultant `registrationState` as the second argument. ] */
+    it('gets the registration id and consequent sas token and registers', function(callback) {
+      var transport = {
+        setSasToken: sinon.spy(),
+      };
+      var security = {
+        createAuthenticationToken: sinon.stub().callsArgWith(1, null, fakeSasToken),
+        getRegistrationId: sinon.stub().callsArgWith(0, null, fakeRegistrationId)
+      };
+      var clientObj = new SymmetricKeyRegistration(fakeProvisioningHost, fakeIdScope, transport, security);
+      clientObj._pollingStateMachine.register = sinon.stub().callsArgWith(1, null, { registrationState: fakeResponse } );
+      clientObj._pollingStateMachine.disconnect = sinon.stub().callsArgWith(0,null);
+      clientObj.register(function(err, response) {
+        assert.isNotOk(err);
+        assert.strictEqual(response, fakeResponse);
+        assert(clientObj._pollingStateMachine.register.calledOnce);
+        assert.strictEqual(clientObj._pollingStateMachine.register.firstCall.args[0].provisioningHost, fakeProvisioningHost);
+        assert.strictEqual(clientObj._pollingStateMachine.register.firstCall.args[0].idScope, fakeIdScope);
+        assert.strictEqual(clientObj._pollingStateMachine.register.firstCall.args[0].registrationId, fakeRegistrationId);
+        assert(security.getRegistrationId.calledOnce);
+        assert(security.createAuthenticationToken.calledOnce);
+        assert(transport.setSasToken.calledOnce);
+        assert.strictEqual(transport.setSasToken.firstCall.args[0], fakeSasToken);
+        callback();
+      });
+    });
+
+    /* Tests_SRS_NODE_DPS_SYMMETRIC_REGISTRATION_06_007: [ If the `getRegistrationId` fails, the `register` shall call the `_callback` with the error. ] */
+    it('fails if getRegisterId fails', function(callback) {
```

Comment:
>getRegisterId [](start = 17, length = 13)

getRegistrationId

Reply:
Fixed.

---
In reply to: [231362009](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231362009) [](ancestors = 231362009)